src/components/google-maps-aerial.js

```diff
@@ -0,0 +1,144 @@
+import { TilesRenderer } from '3d-tiles-renderer';
+import {
+  TilesFadePlugin,
+  TileCompressionPlugin,
+  GLTFExtensionsPlugin,
+  GoogleCloudAuthPlugin
+} from '3d-tiles-renderer/plugins';
+
+console.log('3d-tiles-renderer', TilesRenderer);
+const MathUtils = AFRAME.THREE.MathUtils;
+const DRACOLoader = AFRAME.THREE.DRACOLoader;
+
+if (typeof AFRAME === 'undefined') {
+  throw new Error(
+    'Component attempted to register before AFRAME was available.'
+  );
+}
+
+AFRAME.registerComponent('google-maps-aerial', {
+  schema: {
+    apiToken: { type: 'string', default: '' },
+    latitude: { type: 'number', default: 37.795 }, // SF Ferry Building
+    longitude: { type: 'number', default: -122.394 },
+    minDistance: { type: 'number', default: 500 },
+    maxDistance: { type: 'number', default: 20000 },
+    ellipsoidalHeight: { type: 'number', default: 0 },
+    copyrightEl: { type: 'selector' }
+  },
+
+  init: function () {
+    this.tick = AFRAME.utils.throttleTick(this.tick, 10, this);
+    this.initialized = false;
+    // Initialize tiles
+    this.tiles = new TilesRenderer(
+      'https://tile.googleapis.com/v1/3dtiles/root.json'
+    );
+
+    // Register plugins
+    this.tiles.registerPlugin(
+      new GoogleCloudAuthPlugin({ apiToken: this.data.apiToken })
+    );
+    this.tiles.registerPlugin(new TileCompressionPlugin());
+    this.tiles.registerPlugin(new TilesFadePlugin());
+    this.tiles.registerPlugin(
+      new GLTFExtensionsPlugin({
+        dracoLoader: new DRACOLoader().setDecoderPath(
+          'https://unpkg.com/three@0.153.0/examples/jsm/libs/draco/gltf/'
+        )
+      })
+    );
+
+    // Set location
+    this.tiles.setLatLonToYUp(
+      this.data.latitude * MathUtils.DEG2RAD,
+      this.data.longitude * MathUtils.DEG2RAD
+    );
+
+    // Create a child entity for the height offset
+    const offsetEl = document.createElement('a-entity');
+    offsetEl.object3D.position.y = -this.data.ellipsoidalHeight;
+    offsetEl.object3D.add(this.tiles.group);
+    this.el.appendChild(offsetEl);
+    this.offsetEl = offsetEl;
+
+    // Get camera and renderer
+    this.camera =
+      this.data.cameraEl?.object3D.children[0] ??
+      document.querySelector('a-scene').camera;
+    this.renderer = this.el.sceneEl.renderer;
+    // if (!this.camera) {
+    //   throw new Error('3D Tiles: Please add an active camera or specify the target camera via the cameraEl property');
+    // }
+    this.originalCamera = this.camera;
+
+    this.tiles.setResolutionFromRenderer(this.camera, this.renderer);
+    this.tiles.setCamera(this.camera);
+    this.tiles.update();
+    this.initialized = true;
```

Comment:
It seems that `initialized` flag is completely useless. You can remove it and in all conditions.

Reply:
8ef2dd644a2fd481fbb42fb085e08b420606688c